arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt

```diff
@@ -1496,7 +1496,11 @@ public fun <A, B> Either<A, B>.combine(SGA: Semigroup<A>, SGB: Semigroup<B>, b:
     }
   }
 
+@Deprecated("use fold instead", ReplaceWith("fold(MA, MB)", "arrow.core.fold"))
```

Comment:
Should instead of adding this new fold method deprecate towards `fold(Monoid.either(MA, MB))`?
These overloads seem a bit redundant.

Reply:
I also think these overloads are redundant if we already have one instance for Either that we could take as a single argument. 

I never use `Monoid` if I can avoid it and prefer a method that takes an `empty` value and an `assoc` function. That method already exists for Iterable, and it's just `fold`. In actual use cases in Kotlin, people are not going to provide their own `Monoid` or `Semigroup` instances to pass them around. These instances we provide are mainly for primitive types, but most use cases have more complex models inside the Iterable. Since users would not bother to construct complicated instances for `Monoid` over their model, these functions quickly become useless and need to be rewritten to basic `fold` or similar.